hamocc/mo_read_sedpor.F90

```diff
@@ -0,0 +1,130 @@
+! Copyright (C) 2020  S. Gao, I. Bethke, J. Tjiputra, J. Schwinger
+!
+! This file is part of BLOM/iHAMOCC.
+!
+! BLOM is free software: you can redistribute it and/or modify it under the
+! terms of the GNU Lesser General Public License as published by the Free 
+! Software Foundation, either version 3 of the License, or (at your option) 
+! any later version. 
+!
+! BLOM is distributed in the hope that it will be useful, but WITHOUT ANY 
+! WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS 
+! FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for
+! more details. 
+!
+! You should have received a copy of the GNU Lesser General Public License 
+! along with BLOM. If not, see https://www.gnu.org/licenses/.
+
+module mo_read_sedpor
+!*****************************************************************************
+! Purpose
+! -------
+!   - Routine for reading sediment porosity from netcdf file
+!
+! Description
+! -----------
+! Public routines and variable of this module:
+!
+!   - subroutine ini_read_sedpor
+!        read sediment porosity file
+!
+!   L_SED_POR must be set to true in nml to activate 
+!   lon-lat variable sediment porosity.
+!
+!   The model attempts to read lon-lat-sediment depth variable porosity
+!   from the input file 'SEDPORFILE' (incl. full path)
+!  
+!   sed_por holds then the porosity that can be applied later 
+!   via mo_apply_sedpor
+!
+!*****************************************************************************
+
+implicit none
+
+private
+
+public :: ini_read_sedpor,sedporfile,sed_por
+
+character(len=512),save :: sedporfile = ''
+real,save,allocatable   :: sed_por(:,:,:) ! holds input data from nc-file 
+
+
+contains
+
+subroutine ini_read_sedpor(kpie,kpje,ks,omask)
```

Comment:
Actually, there is no need to allocate the array sed_por permanently: The array is only needed to transfer the data from this routine to bodensed (both called from hamocc_ini). So one could define sed_por as a local (automatic) array in hamocc_ini, pass it first to read_sedpor and then to bodensed. On return from hamocc_ini, sed_por will be automatically deallocated. I would then also rename this routine to `read_sedpor` because then there is nothing that is really initialised (this would be more consistent with the fact that this is a one time read operation, not something that is used later on).

On a side note, one could integrate `bodensed` into `mo_sedmnt` and rename it to `ini_sedmnt`, which would make more sense than having it as a stand-alone subroutine. Maybe something for later.